code/scripts/espurna_utils/bsec.py

```diff
@@ -1,74 +1,58 @@
 import os
-import pathlib
+from SCons.Script import Move, Delete, Mkdir, Touch
+
 
 def bsec_inject_patcher(env):
-    bsec_builder = next((builder for builder in env.GetLibBuilders() if builder.name == 'BSEC Software Library'), None)
+    bsec_builder = next(
+        (
+            builder
+            for builder in env.GetLibBuilders()
+            if builder.name == "BSEC Software Library"
+        ),
+        None,
+    )
 
     if bsec_builder is None:
         return
 
-    bsec_dir = os.path.join(bsec_builder.src_dir, 'esp8266')
-    patchflag_path = os.path.join(bsec_dir, '.patched')
+    bsec_dir = os.path.join(bsec_builder.src_dir, "esp8266")
 
-    env.Append(LIBPATH=[bsec_dir])
-    env.Append(LIBS=["algobsec"])
+    out_path = os.path.join(bsec_dir, "out")
+    patchflag_path = os.path.join(bsec_dir, ".patched")
 
-    patch_action = env.Command(
-        os.path.join(bsec_dir, "libalgobsec.a.out"),
-        os.path.join(bsec_dir, "libalgobsec.a"),
+    # Note:
+    # depending on libalgobsec.a or libalgobsec.a.orig will force SCons to always rebuild
+    # this happens b/c last time it had seen it and calculated deps, there was still the original .a
+    patcher = env.Command(
+        patchflag_path,
+        None,
```

Comment:
btw if you happen on SCons again with PIO or otherwise, `--debug=explain` is kind-of helpful here.
for PIO we need to patch it, as it seems though, idk if SCons accepts some env variable as configuration:
```diff
diff --git a/platformio/managers/platform.py b/platformio/managers/platform.py
index ada4f4ac..2fe210a8 100644
--- a/platformio/managers/platform.py
+++ b/platformio/managers/platform.py
@@ -449,6 +449,7 @@ class PlatformRunMixin(object):
             proc.get_pythonexe_path(),
             join(get_core_package_dir("tool-scons"), "script", "scons"),
             "-Q",
+            "--debug=explain",
             "--warn=no-no-parallel-support",
             "--jobs",
             str(jobs),
```

Reply:
This is really useful. Would make sense to passthrough `-v` from PlatformIO here right?

Reply:
`--scons-args='...'`? Or something like that

Reply:
Or that.. but asking for a verbose build could imply this as well.

Reply:
idk, we'd still need to ask PIO foks:
https://github.com/platformio/platformio-core/issues

For --verbose it is additional line of log after each .o build, too, so things get cluttered. I really see it as an additional feature for scons arguments specifically, but the main question is how exactly this --argument proxy should workq when we already have an app accepting arguments